tests/libwallet_api_tests/main.cpp

```diff
@@ -203,10 +327,291 @@ TEST_F(WalletManagerTest, WalletManagerStoresWallet3)
+    history->refresh();
+    ASSERT_TRUE(history->count() > 0);
+    size_t count1 = history->count();
+
+    std::cout << "**** Transactions before transfer (" << count1 << ")" << std::endl;
+    for (auto t: history->getAll()) {
+        ASSERT_TRUE(t != nullptr);
+        Utils::print_transaction(t);
+    }
+
+    std::string wallet4_addr = Utils::get_wallet_address(CURRENT_DST_WALLET, TESTNET_WALLET_PASS);
+
+    Bitmonero::PendingTransaction * tx = wallet_src->createTransaction(wallet4_addr, AMOUNT_10XMR * 5, 0);
+    ASSERT_TRUE(tx->status() == Bitmonero::PendingTransaction::Status_Ok);
+    ASSERT_TRUE(tx->commit());
+    history = wallet_src->history();
+    history->refresh();
+    ASSERT_TRUE(count1 != history->count());
+
+    std::cout << "**** Transactions after transfer (" << history->count() << ")" << std::endl;
+    for (auto t: history->getAll()) {
+        ASSERT_TRUE(t != nullptr);
+        Utils::print_transaction(t);
+    }
+}
+
+struct MyWalletListener : public Bitmonero::WalletListener
+{
+
+    Bitmonero::Wallet * wallet;
+    uint64_t total_tx;
+    uint64_t total_rx;
+    std::timed_mutex  guard;
+
+    MyWalletListener(Bitmonero::Wallet * wallet)
+        : total_tx(0), total_rx(0)
+    {
+        this->wallet = wallet;
+        this->wallet->setListener(this);
+    }
+
+    virtual void moneySpent(const string &txId, uint64_t amount)
+    {
+        std::cout << "wallet: " << wallet->address() << " just spent money ("
+                  << txId  << ", " << wallet->displayAmount(amount) << ")" << std::endl;
+        total_tx += amount;
+        guard.unlock();
+    }
+
+    virtual void moneyReceived(const string &txId, uint64_t amount)
+    {
+        std::cout << "wallet: " << wallet->address() << " just received money ("
+                  << txId  << ", " << wallet->displayAmount(amount) << ")" << std::endl;
+        total_rx += amount;
+        guard.unlock();
+    }
+};
+
+/*
+TEST_F(WalletTest2, WalletCallbackSent)
+{
+
+    Bitmonero::Wallet * wallet_src = wmgr->openWallet(TESTNET_WALLET3_NAME, TESTNET_WALLET_PASS, true);
+    // make sure testnet daemon is running
+    ASSERT_TRUE(wallet_src->init(TESTNET_DAEMON_ADDRESS, 0));
+    ASSERT_TRUE(wallet_src->refresh());
+    MyWalletListener * wallet_src_listener = new MyWalletListener(wallet_src);
+    std::cout << "** Balance: " << wallet_src->displayAmount(wallet_src->balance()) <<  std::endl;
+
+
+    uint64_t amount = AMOUNT_10XMR * 5;
+    std::cout << "** Sending " << Bitmonero::Wallet::displayAmount(amount) << " to " << TESTNET_WALLET4_ADDRESS;
+    Bitmonero::PendingTransaction * tx = wallet_src->createTransaction(TESTNET_WALLET4_ADDRESS, AMOUNT_1XMR * 5);
+    ASSERT_TRUE(tx->status() == Bitmonero::PendingTransaction::Status_Ok);
+    ASSERT_TRUE(tx->commit());
+
+    std::chrono::seconds wait_for = std::chrono::seconds(60*3);
+
+    wallet_src_listener->guard.lock();
+    wallet_src_listener->guard.try_lock_for(wait_for);
```

Comment:
this is simply "wait for money sent for no more than 3 minutes".  guard.unlock() called in MyWalletListener, while the "monitoring thread" is not yet implemented in Wallet class.